From 6220d7ef31a6886297c2c839ec93a547f13a0a79 Mon Sep 17 00:00:00 2001
From: Nancy Chen <Nancy.Chen@freescale.com>
Date: Fri, 23 Jan 2009 11:09:39 -0600
Subject: [PATCH] ENGR00106138 MX37: Move CPU voltage changes for MC13892

MX37: Move CPU voltage changes for MC13892 to mc13892_reg_int()

Signed-off-by: Nancy Chen <Nancy.Chen@freescale.com>
---
 arch/arm/mach-mx37/mx37_3stack.c |   24 +++++++++++-------------
 1 files changed, 11 insertions(+), 13 deletions(-)

diff --git a/arch/arm/mach-mx37/mx37_3stack.c b/arch/arm/mach-mx37/mx37_3stack.c
index 6a408e9..16f324a 100644
--- a/arch/arm/mach-mx37/mx37_3stack.c
+++ b/arch/arm/mach-mx37/mx37_3stack.c
@@ -73,7 +73,7 @@ static struct cpu_wp cpu_wp_auto[] = {
 	 .mfi = 5,
 	 .mfd = 23,
 	 .mfn = 13,
-	 .cpu_voltage = 1050000,},
+	 .cpu_voltage = 1000000,},
 	{
 	 .pll_rate = 400000000,
 	 .cpu_rate = 400000000,
@@ -81,7 +81,7 @@ static struct cpu_wp cpu_wp_auto[] = {
 	 .mfi = 8,
 	 .mfd = 2,
 	 .mfn = 1,
-	 .cpu_voltage = 950000,},
+	 .cpu_voltage = 900000,},
 	{
 	 .pll_rate = 200000000,
 	 .cpu_rate = 200000000,
@@ -89,7 +89,7 @@ static struct cpu_wp cpu_wp_auto[] = {
 	 .mfi = 8,
 	 .mfd = 2,
 	 .mfn = 1,
-	 .cpu_voltage = 900000,},
+	 .cpu_voltage = 850000,},
 	{
 	 .pll_rate = 600000000,
 	 .cpu_rate = 600000000,
@@ -97,7 +97,7 @@ static struct cpu_wp cpu_wp_auto[] = {
 	 .mfi = 6,
 	 .mfd = 3,
 	 .mfn = 1,
-	 .cpu_voltage = 1250000,},
+	 .cpu_voltage = 1200000,},
 };
 
 struct cpu_wp *get_cpu_wp(int *wp)
@@ -110,6 +110,8 @@ static void mc13892_reg_int(void)
 {
 	int i = 0;
 	struct regulator *regulator;
+	struct cpu_wp *cpu_wp_tbl1;
+	int cpu_wp_nr1;
 	char *reg_name[] = {
 		"SW1",
 		"SW2",
@@ -163,6 +165,11 @@ static void mc13892_reg_int(void)
 			regulator_put(regulator, NULL);
 		}
 	}
+
+	/* Set the current working point. */
+	cpu_wp_tbl1 = get_cpu_wp(&cpu_wp_nr1);
+	for (i = 0; i < cpu_wp_nr1; i++)
+		cpu_wp_tbl1[i].cpu_voltage += 50000;
 }
 
 late_initcall(mc13892_reg_int);
@@ -812,15 +819,6 @@ static inline void mxc_sgtl5000_init(void)
  */
 static void mx37_3stack_fixup_for_board_v1(void)
 {
-	struct cpu_wp *cpu_wp_tbl1;
-	int cpu_wp_nr1;
-	int i;
-
-	/* Set the current working point. */
-	cpu_wp_tbl1 = get_cpu_wp(&cpu_wp_nr1);
-	for (i = 0; i < cpu_wp_nr1; i++)
-		cpu_wp_tbl1[i].cpu_voltage -= 50000;
-
 	dptc_gp_data.reg_id = "DCDC1";
 	dptc_lp_data.reg_id = "DCDC4";
 	gp_reg_id = "DCDC1";
-- 
1.5.4.4

